models/issue_label.go

```diff
@@ -364,11 +492,54 @@ func UpdateLabel(l *Label) error {
 	return updateLabel(x, l)
 }
 
-// DeleteLabel delete a label of given repository.
-func DeleteLabel(repoID, labelID int64) error {
-	_, err := GetLabelInRepoByID(repoID, labelID)
+// DeleteLabel delete a label of given repository or organiization.
+func DeleteLabel(id, labelID int64, isOrg bool) error {
+	var err error
+	if isOrg {
+		_, err = GetLabelInOrgByID(id, labelID)
+		if err != nil {
+			if IsErrRepoLabelNotExist(err) {
+				return nil
+			}
+			return err
+		}
+	} else {
+		_, err = GetLabelInRepoByID(id, labelID)
+		if err != nil {
+			if IsErrRepoLabelNotExist(err) {
+				return nil
+			}
+			return err
+		}
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err = sess.Begin(); err != nil {
+		return err
+	}
+
+	if _, err = sess.ID(labelID).Delete(new(Label)); err != nil {
+		return err
+	} else if _, err = sess.
+		Where("label_id = ?", labelID).
+		Delete(new(IssueLabel)); err != nil {
+		return err
+	}
+
+	// Clear label id in comment table
```

Comment:
Perhaps we should delete the comment instead. These are comments type `CommentTypeLabel` with no other useful content (just the string "1").
https://github.com/go-gitea/gitea/blob/13fadae071770d1ec330bc413aacfe20be73d9a4/models/issue_label.go#L606-L616

Reply:
Hmm strange this isn't actually my code or from this PR but it seems included perhaps because the function has moved down far enough from where it originally was. My code is just the check on top for repo or org and adding the isOrg bool to the expected params.

Reply:
Perhaps my comment was a little confusing. The code I meant to be reviewing:

![image](https://user-images.githubusercontent.com/18600385/77805957-1767b580-7062-11ea-8044-09167dbda1f2.png)

The snippet I pasted in my comment before was to exemplify the content of a comment row regarding to label assignment.

Instead of `.Update` and setting the `label_id` to 0, we should be using `.Delete` and remove the row entirely, I believe.

Reply:
OK cool I'll look into it -- I just meant that the above code isn't from this PR and already exists so I'm not familiar with it but since I'm making a lot of changes anyway : )

Reply:
OK fixed to delete comment(s) when label is deleted